addons/point_of_sale/static/src/js/db.js

```diff
@@ -283,38 +287,47 @@ var PosDB = core.Class.extend({
             }
             if (!this.partner_by_id[partner.id]) {
                 this.partner_sorted.push(partner.id);
+            } else {
+                const oldPartner = this.partner_by_id[partner.id];
+                if (oldPartner.barcode || partner.barcode) {
+                    delete this.partner_by_barcode[oldPartner.barcode];
+                    this.partner_by_barcode[partner.barcode] = partner;
```

Comment:
what if the added partner has no barcode anymore ?

Reply:
good catch